### ISSUE TITLE
Refactor Tests

### ISSUE THREAD

**Issue by casey** [agora-org/agora#258]
Once #253 lands, we'll have a mix of tests that run from the same process, and tests that run in a separate process, with a goal to move all tests to run in a separate process.

We originally started moving the tests as-is, and writing support code `fn test`, `TestContext`, etc, to allow moving the tests without modifying them.

I found this to be pretty confusing though. We have a bunch of different functions that wrap each other and set different things (nothing, args, and environment, which can carry a bunch of configuration), and it's hard to get moving them right. I tried to move `test_with_environment` over, and was pretty quickly confused at how to make this work in terms of the existing tests.

I think it might actually be easier for me to refactor the tests in `integration.rs`, and then start moving the remaining tests over, modifying them as I move them.

I would refactor the tests to use builder that allows customizing the test environment, use blocking I/O, and be in a more imperative style. Something like:

```
#[test]
fn test_foo() {
   let agora = AgoraInstance::builder()
    .http_port(1337)
    .build();

  agora.write("files", "some precious content");

  let text = (agora.files_url().join("foo"));

  assert_eq!(text, "some precious content");
}
```

It's not ideal to modify the tests as we move them, but it's easier for me to think about having a builder which customizes the test environment and returns a test instance, which is then interrogated for the test, than the current nesting functions, which I just found really confusing when migrating tests.

What do you think? Definitely open to feedback on this.

**Comment by soenkehahn** [comment]
I think it's up to you, if you think that's easier, by all means go ahead.

What you could also do is first refactoring the tests in place to make them easier to move. And then move them in a separate PR. But yeah, up to you.
